### PR TITLE
case LILY-30: Dockable Interactive Windows

### PR DESCRIPTION
**Pull request by danteruiz**
Dockable Interactive Windows

https://highfidelity.atlassian.net/projects/LILY/issues/LILY-30

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19183/HighFidelity-Beta-PR15507-fe46f66.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19183/HighFidelity-Quest-Beta-PR15507-fe46f66.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19188/HighFidelity-Beta-PR15507-7d52e6e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19188/HighFidelity-Quest-Beta-PR15507-7d52e6e.apk)

**Comment by danteruiz** [comment]
Test script 
```
(function() {
    var window = Desktop.createWindow(Script.resourcesPath() + "qml/hifi/Test.qml", { dock: true });
}());```

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19193/HighFidelity-Beta-PR15507-759e905.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19193/HighFidelity-Beta-Interface-PR15507-759e905.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19193/HighFidelity-Beta-PR15507-759e905.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19193/HighFidelity-Quest-Beta-PR15507-759e905.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19193/HighFidelity-Beta-PR15507-759e905.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19193/HighFidelity-Beta-Interface-PR15507-759e905.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19227/HighFidelity-Beta-PR15507-7ac8a5a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19227/HighFidelity-Beta-Interface-PR15507-7ac8a5a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19227/HighFidelity-Beta-PR15507-7ac8a5a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19227/HighFidelity-Quest-Beta-PR15507-7ac8a5a.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19227/HighFidelity-Beta-PR15507-7ac8a5a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19227/HighFidelity-Beta-Interface-PR15507-7ac8a5a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19269/HighFidelity-Beta-PR15507-faf0799.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19269/HighFidelity-Beta-Interface-PR15507-faf0799.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19269/HighFidelity-Beta-PR15507-faf0799.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19269/HighFidelity-Beta-Interface-PR15507-faf0799.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19269/HighFidelity-Beta-PR15507-faf0799.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19269/HighFidelity-Quest-Beta-PR15507-faf0799.apk)

File: libraries/ui/src/DockWidget.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//  libraries/ui/src
+//
+//  Created by Dante Ruiz 05-07-2019
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#pragma once
```
Comment (samcake) [inline libraries/ui/src/DockWidget.h]: We need a real header file protection, not only pragma once

File: interface/resources/qml/hifi/Test.qml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import QtQuick 2.0
```
[the remark this answers is not in the document]
Comment (danteruiz) [reply interface/resources/qml/hifi/Test.qml]: I deleted that file

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19271/HighFidelity-Beta-PR15507-3806631.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19271/HighFidelity-Beta-Interface-PR15507-3806631.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19271/HighFidelity-Beta-PR15507-3806631.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19271/HighFidelity-Beta-Interface-PR15507-3806631.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19271/HighFidelity-Beta-PR15507-3806631.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19271/HighFidelity-Quest-Beta-PR15507-3806631.apk)

**Pull request merged**